src/app/(sidebar)/smart-contracts/deploy-contract/page.tsx

```diff
@@ -396,392 +493,617 @@ export default function DeployContract() {
+                          size="sm"
+                          weight="semi-bold"
+                          addlClassName="DeployContract__darkHeading"
+                        >
+                          {CONSTRUCTOR_KEY}
+                        </Text>
+
+                        <JsonSchemaRenderer
+                          formError={constructorFormError}
+                          setFormError={setConstructorFormError}
+                          name={CONSTRUCTOR_KEY}
+                          schema={constructorSchema as JSONSchema7}
+                          onChange={(val) => {
+                            setConstructorFormValue(val);
+                            setIsConstructorArgsFilled(isArgsFilled(val.args));
+                          }}
+                          parsedSorobanOperation={constructorFormValue}
+                        />
+                      </Box>
+                    </Card>
+                  ) : null}
+
+                  {deployTxError ? (
+                    <Notification
+                      title="Deploy transaction error"
+                      variant="error"
+                    >
+                      {`There was an error building the Deploy transaction: ${deployTxError.toString()}`}
+                    </Notification>
+                  ) : null}
+
+                  <Box gap="sm" direction="row" justify="end">
+                    <Button
+                      variant="secondary"
+                      size="md"
+                      isLoading={isDeployTxLoading || isDeployTxFetching}
+                      disabled={getIsBuildDeployButtonDisabled()}
+                      onClick={async () => {
+                        const wasmHashBytes = getWasmHashBytes();
+
+                        if (sourceAccount && wasmHashBytes) {
+                          const operation = Operation.createCustomContract({
+                            wasmHash: wasmHashBytes,
+                            address: Address.fromString(sourceAccount),
+                            constructorArgs: getOrderedConstructorArgs(
+                              constructorFormValue.args,
+                              constructorSchema,
+                            ),
+                          });
+
+                          setDeployOp(operation);
+                        }
+                      }}
+                    >
+                      Build Deploy transaction
+                    </Button>
+                  </Box>
+
+                  <SignTransactionXdr
+                    key={`deploy-tx-${signTxResetKey}`}
+                    id="deploy-tx"
+                    title="Add signature to deploy"
+                    xdrToSign={deployTx?.preparedXdr || null}
+                    onDoneAction={({
+                      signedXdr,
+                      successMessage,
+                      errorMessage,
+                    }) => {
+                      setSignedDeployTx(signedXdr);
```

Comment:
Thoughts on using `useReducer` for 

```setSignedDeployTx(signedXdr);
setSignDeploySuccess(successMessage);
setSignDeployError(errorMessage);```

Reply:
I don't have much experience using `useReducer`. This state is simple, and we're assigning values that we get back from the request. I don't know if using action here instead simplifies the logic (I would prefer keeping the state simple and not using an object). Plus, the overhead of setting up the reducer and actions. Do you think it would be better to use `useReducer` here?